examples/cpp/RK2.cpp

```diff
@@ -0,0 +1,77 @@
+#include "mole.h"  
+#include <iostream>
+#include <sstream> 
+#include <fstream>  // For file output
+#include <cmath>    // For sin()
+#include <iomanip>  // For setprecision
+
+// Function prototype for f(t, y)
+double f(double t, double y);
+
+int main() {
+    constexpr double h = 0.1;       // Step size
+    constexpr double t_start = 0.0; // Initial time
+    constexpr double t_end = 5.0;   // Final time
+
+    int n_steps = static_cast<int>((t_end - t_start) / h) + 1;
+
+    // MOLE's vec type for vectors
+    vec t(n_steps);              // Time vector
+    vec y(n_steps);              // Solution vector
+
+    // Initial conditions
+    t(0) = t_start;
+    y(0) = 2.0; 
+
+    // Populate the time vector
+    for (int i = 1; i < n_steps; ++i) {
+        t(i) = t(i - 1) + h;
+    }
+
+    // RK2 Method
+    for (int i = 0; i < n_steps - 1; ++i) {
+        const double k1 = f(t(i), y(i));                              // Slope at the beginning
+        const double k2 = f(t(i) + h / 2.0, y(i) + h / 2.0 * k1);     // Slope at midpoint
+        y(i + 1) = y(i) + h * k2;                                    // Update solution
+    }
+
+    // Write data to a file
+    std::ofstream data_file("data.txt");
+    if (!data_file) {
+        std::cerr << "Error: Failed to open data file.\n";
+        return 1;
+    }
```

Comment:
I would not dump values to file, but to screen (standard output). 

However, you want to make sure that your C++ example exactly reproduces (or up to a good chunk of significant digits) the MATLAB example. So dumping values to file is a good debugging/verification practice. You can then do the same thing with the corresponding MATLAB file and compare the numerical values for correctness. 

However, in the version pushed to the PR, I would prefer to print out to standard output. Thank you!